### PR TITLE
[dev] PBI AB#80783 - Widgetize Connection Forms.

### PR DESCRIPTION
**Pull request by War777**
This PR contains the following updates for [PBI AB#80783](https://dev.azure.com/saddlebackchurch/Church%20Management/_workitems/edit/80783) - Widgetize Connection Forms:

- Enables the `<Grid />` responsiveness interaction to addapt with either the browser or the parent container width.
- Add SASS file with @container queries support (Our `JSS` version/implementation doesn't do).
- Exposes the **Material-UI's** `createGenerateClassName` function so we can override the resultant `JSS` `classNames` prefixes, previosly generated as `jss1`, `jss2`, ... ,`jssN` and now `connection-form1`, `connection-form2`, ..., `connection-formN`.
This was provoking conflicts when getting two widgets in the same page, now the `className` prefixes interact with the widget type (**connection-form**, **giving-checkout**).

### CODE REVIEW

**Comment by War777** [comment]
**do not merge** label added because I found a bug while doing integrated test, hoping to get it fixed before the end of the day, thanks for the patience.

File: src/layout/grid/grid.jsx [hunk 1]
```diff
@@ -330,6 +338,24 @@ const Grid = React.forwardRef(
             },
         );
 
+        if (responsiveQueryType === 'container') {
+            console.log('pre gridColumns', children);
+        }
```
Comment (groberts314) [inline src/layout/grid/grid.jsx:343]: Will this go away when you're done debugging?

File: src/layout/grid/grid.jsx [hunk 2]
```diff
@@ -330,6 +338,24 @@ const Grid = React.forwardRef(
...
+                        ...child.props,
+                        responsiveQueryType: 'container',
+                    },
+                }),
+            ) :
+            children;
+
+        if (responsiveQueryType === 'container') {
+            console.log('gridColumns', gridColumns);
+        }
```
Comment (groberts314) [inline src/layout/grid/grid.jsx:358]: Same here: please remove when you're done debugging

File: src/layout/grid/gridColumn.jsx [hunk 3]
```diff
@@ -332,27 +335,38 @@ const GridColumn = React.forwardRef(
...
             style,
             xl,
+            responsiveQueryType,
             ...otherProps
         } = props;
 
         const classes = useStyles(props);
 
+        const hasContainerQueriesEnabled = responsiveQueryType === 'container';
+        console.log('hasContainerQueriesEnabled', hasContainerQueriesEnabled);
```
Comment (groberts314) [inline src/layout/grid/gridColumn.jsx:345]: 1) Please remove `console.log()` when done debugging.
2) Please leave Line 345 blank (empty line between declaration on 344 and the multiline one on 346).

File: src/layout/grid/gridColumn.scss [hunk 4]
```diff
@@ -0,0 +1,689 @@
...
+        flex-basis: 91.666667%;
+        max-width: 91.666667%;
+        width: 91.666667%;
+    };
+    .grid--column-desktop-large-12 {
+        flex-basis: 100%;
+        max-width: 100%;
+        width: 100%;
+    }
+}
```
Comment (groberts314) [inline src/layout/grid/gridColumn.scss:689]: Please end in a blank line

**Review by groberts314 — COMMENTED** [review]
Looking okay so far.
I'm definitely curious about the CSS story for widgets.
Some `console.log()`s to cleanup when the debugging is finished.

File: src/layout/grid/grid.jsx [hunk 5]
```diff
@@ -118,6 +118,8 @@ const propTypes = {
      * Deprecated prop. Please use `classes` to override styles.
      */
     verticalAlign: PropTypes.oneOf(['bottom', 'middle', 'top']),
+
+    responsiveQueryType: PropTypes.oneOf(['media', 'container']),
```
Comment (fantaJinMode) [inline src/layout/grid/grid.jsx:122]: **NAB** : Would be great, if we can add a single line description indicating what this prop would do?

File: src/layout/grid/gridColumn.jsx [hunk 6]
```diff
@@ -128,6 +128,8 @@ const propTypes = {
      * Deprecated prop. Please use `classes` to override styles.
      */
     width: PropTypes.oneOf(['auto', true, false, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),
+
+    responsiveQueryType: PropTypes.oneOf(['media', 'container']),
```
Comment (fantaJinMode) [inline src/layout/grid/gridColumn.jsx:132]: **NAB**: Would be great, if we can add a single line description indicating what this prop would do?

**Review by fantaJinMode — COMMENTED** [review]
LGTM 👍  Few NAB comments and agreed with @groberts314  on removing debugging codes.

File: src/layout/grid/grid.jsx (hunk 1 above)
Comment (War777) [reply src/layout/grid/grid.jsx]: Remove!

File: src/layout/grid/grid.jsx (hunk 2 above)
Comment (War777) [reply src/layout/grid/grid.jsx]: Removed!

File: src/layout/grid/gridColumn.scss (hunk 4 above)
Comment (War777) [reply src/layout/grid/gridColumn.scss]: Done!

File: src/layout/grid/gridColumn.jsx (hunk 6 above)
Comment (War777) [reply src/layout/grid/gridColumn.jsx]: Done!

**Comment by groberts314** [comment]
@War777 -- heads up
* https://github.com/saddlebackdev/react-cm-ui/pull/437#discussion_r1176702963
* Please remove **`do not merge`** label when you're ready

**Review by groberts314 — APPROVED** [review]

**Comment by groberts314** [comment]
Last remaining concern is ensuring new styles for the Grid are included in overall CSS in the React CM UI package.  @War777 is checking into this (he was seeing some anomalous behavior with local testing).  Holding off merging this until @War777 confirms all is good.

**Comment by groberts314** [comment]
Added **`WIP`** label.  @War777 is working on a solution for the `@container` query problem.

**Comment by War777** [comment]
I just added the workaround for @container queries but I still need to do some integration testing in the back office side, hopefully wrapping all up this afternoon (I'm still working on creating the firts PR for General Ledger Finance project).

**Comment by groberts314** [comment]
This is pending some integrated testing from @War777.  Anticipate that can be done tomorrow (2023-05-12) in the morning.

File: package.json
[the remark this answers is not in the document]
Comment (groberts314) [reply package.json:1]: Done in bb3602c.

File: src/style.scss [hunk 7]
```diff
@@ -43,6 +43,9 @@
 @import './inputs/timePicker/timePicker';
 
 // Layout
+/* disable-gulp-start */
+@import './layout/grid/gridColumn';
+/* disable-gulp-end */
```
[the remark this answers is not in the document]
Comment (groberts314) [reply src/style.scss:50]: Done: https://github.com/saddlebackdev/react-cm-ui/pull/437/commits/d537343d9330072b1af20c97a8462874c572e7d5#diff-c0a5ddd5b365322730802e8cb15c95c3b6e965cbd3e569c69a2f089753844cdbR46-R47

**Comment by groberts314** [comment]
Closing the PR for now but retaining the feature branch.
It seems this capability ( `@container` queries in CSS/SCSS/JSS) is not required, given current widget usage.
We may want to pursue this capability at some future point (which is why we're keeping the branch), but hopefully then too, the tooling might have proper support and some of the hacks in here might not be required.

**Pull request closed without merging**